server/graphql/v2/query/collection/ExpensesCollectionQuery.ts

```diff
@@ -396,12 +401,43 @@ export const ExpensesCollectionQueryResolver = async (
   } else if (args.tag === null || args.tags === null) {
     where['tags'] = { [Op.is]: null };
   }
-  if (args.minAmount) {
-    where['amount'] = { [Op.gte]: args.minAmount };
-  }
-  if (args.maxAmount) {
-    where['amount'] = { ...where['amount'], [Op.lte]: args.maxAmount };
+  if (args.minAmount || args.maxAmount) {
+    console.log(args);
+    const operator =
+      args.minAmount && args.maxAmount
+        ? { [Op.between]: [args.minAmount, args.maxAmount] }
+        : args.minAmount
+          ? { [Op.gte]: args.minAmount }
+          : { [Op.lte]: args.maxAmount };
+
+    if (args.amountCurrency) {
```

Comment:
If `amountCurrency` is not provided but we're still filtering on `minAmount`/`maxAmount`, I would still default to a given currency (the host's?), to make sure we resolve the issue consistently, even for people not using this new flag.